python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -1147,32 +1150,32 @@ def serialize(self) -> str:
             {
                 "version": self.SERIALIZATION_VERSION,
                 # sort to ensure that equivalent partition subsets have identical serialized forms
-                "subset": sorted(list(self._subset)),
+                "subset": sorted(list(self.subset)),
```

Comment:
Do we  need to make sure that the default serializer serializes this in sorted order? cc @OwenKephart

Reply:
ah yeah we do, this is theoretically under test at the moment: https://sourcegraph.com/github.com/dagster-io/dagster/-/blob/python_modules/dagster/dagster_tests/core_tests/partition_tests/test_partition.py?L154

Reply:
ah I see, that test would need to be updated in order to test the new behavior (as we'd need to go down the `deserialize_value(serialize_value(...))` route rather than manually calling `.serialize()`.

cc @clairelin135 to add a test along these lines -- my guess is that it may fail, but worst case just need a custom field serializer to enforce the sorting there

Reply:
added a test, looks like we already do sorting in serdes: https://github.com/dagster-io/dagster/blob/claire/default-subset-serialization/python_modules/dagster/dagster/_serdes/serdes.py#L712-L714

Reply:
ah nice!